src/main/java/com/google/sps/data/RentPost.java

```diff
@@ -0,0 +1,152 @@
+package com.google.sps;
+
+import static com.google.sps.Constants.CURRENCY_CODE;
+import static com.google.sps.Constants.DATE_FORMAT;
+import static com.google.sps.Constants.RENT_COLLECTION_NAME;
+import static com.google.sps.Constants.REQUEST_DESCRIPTION;
+import static com.google.sps.Constants.REQUEST_END_DATE;
+import static com.google.sps.Constants.REQUEST_LEASE_TYPE;
+import static com.google.sps.Constants.REQUEST_NUM_ROOMS;
+import static com.google.sps.Constants.REQUEST_PRICE;
+import static com.google.sps.Constants.REQUEST_ROOM_TYPE;
+import static com.google.sps.Constants.REQUEST_START_DATE;
+import static com.google.sps.Constants.REQUEST_TIMESTAMP;
+import static com.google.sps.Constants.REQUEST_TITLE;
+
+import com.google.auto.value.AutoValue;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.Timestamp;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.money.CurrencyUnit;
+import javax.money.Monetary;
+import org.javamoney.moneta.Money;
+
+/** A rent post made by a user. */
+@AutoValue
+public abstract class RentPost {
+  enum LeaseType {
+    YEAR_LONG, 
+    MONTH_TO_MONTH, 
+    SUMMER_SUBLET;
+  }
+  enum RoomType {
+    SINGLE,
+    SHARED;
+  }
+
+  public abstract String description();
+  @Nullable public abstract Date endDate();
+  public abstract LeaseType leaseType();
+  public abstract int numRooms();
+  public abstract Money price();
+  public abstract RoomType roomType();
+  @Nullable public abstract Date startDate();
+  public abstract String title();
+
+  public static Builder builder() {
+    return new AutoValue_RentPost.Builder();
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    abstract Builder setDescription(String description);
+    abstract Builder setEndDate(Date endDate);
+    abstract Builder setLeaseType(LeaseType leaseType);
+    abstract Builder setNumRooms(int numRooms);
+    abstract Builder setPrice(Money price);
+    abstract Builder setRoomType(RoomType roomType);
+    abstract Builder setStartDate(Date startDate);
+    abstract Builder setTitle(String title);
+
+    abstract LeaseType leaseType();
+    abstract RoomType roomType();
+    abstract int numRooms();
+
+    abstract RentPost build();
+    
+    public Builder setLeaseType(String leaseType) {
+      setLeaseType(LeaseType.valueOf(leaseType));
```

Comment:
What if the leasType is not from list? We should have argument checker for it and error out if things are invalid.

Reply:
valueOf throws an exception if it is not from the lease. Added functionality to catch the exception in the servlet call and send an error message there.